src/Aspire.Hosting/Publishing/ManifestPublisher.cs

```diff
@@ -132,13 +132,26 @@ private static void WriteExecutable(ExecutableResource executable, ManifestPubli
         context.Writer.WriteString("workingDirectory", relativePathToProjectFile);
 
         context.Writer.WriteString("command", executable.Command);
-        context.Writer.WriteStartArray("args");
 
-        foreach (var arg in executable.Args ?? [])
+        var args = new List<string>(executable.Args ?? []);
+        if (executable.TryGetAnnotationsOfType<ExecutableArgsCallbackAnnotation>(out var argsCallback))
         {
-            context.Writer.WriteStringValue(arg);
+            foreach (var callback in argsCallback)
+            {
+                callback.Callback(args);
+            }
+        }
+
+        if (args.Count > 0)
+        {
+            context.Writer.WriteStartArray("args");
+
+            foreach (var arg in args ?? [])
```

Comment:
`args` will never be null here now right, so you can avoid the `??`

Reply:
Done.